tests/admin_views/tests.py

```diff
@@ -8928,3 +8928,19 @@ def test_non_admin_url_404_if_not_authenticated(self):
         response = self.client.get(unknown_url)
         # Does not redirect to the admin login.
         self.assertEqual(response.status_code, 404)
+
+
+@override_settings(ROOT_URLCONF="admin_views.urls")
+class BaseTemplateBlocks(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.superuser = User.objects.create_superuser(
+            username="super1", password="secret", email="super2@example.com"
+        )
+
+    def setUp(self):
+        self.client.force_login(self.superuser)
+
+    def test_extra_body_block(self):
+        response = self.client.get(reverse("block_view"))
+        self.assertContains(response, "Test string for the extra body block")
```

Comment:
I think we can maybe use `AdminCustomTemplateTests` and add `extrabody` to `tests/admin_views/templates/admin/base_site.html` (then we don't need a new template file and view etc).
I also think it would be nice that the assert shows that this is added just before the closing body tag so adding that in here would be nice 👍